include/mbedtls/aes.h

```diff
@@ -122,6 +122,8 @@ void mbedtls_aes_free( mbedtls_aes_context *ctx );
  *
  * \return         \c 0 on success.
  * \return         #MBEDTLS_ERR_AES_INVALID_KEY_LENGTH on failure.
+ * \return         #MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE if the
+ *                 alternative implementation doesn't support the key size.
```

Comment:
What other reason for a HW acceleration not to support a feature when calling `mbedtls_aes_setkey_enc()`? 
The same goes for other operations. I tried to be as specific as possible, so we know what to test and what not

Reply:
We as a library can't know the many ways in which hardware could choose not to support a certain feature. We are bound to get things wrong for some hardware. I'd rather us try to avoid specifics as much as possible, to give hardware freedom to err however they see fit.

Reply:
But we can define what is mandatory and what not. A developer that wants to develop and alternative implementation will need some guidance on what to return when.

Reply:
OK